src/metaopt/io/database/mongodb.py

```diff
@@ -108,33 +105,26 @@ def update(self, collection_name, update_dictionary, field_filters=None):
         }
         """
 
-        if field_filters is None:
-            field_filters = {}
-
         dbcollection = self.db[collection_name]
 
         # see http://api.mongodb.com/python/current/api/pymongo/collection.html
         # for more details
 
-        result = dbcollection.update_one(field_filters, update_dictionary)
+        result = dbcollection.update_one(query, update_dictionary)
 
         return result
 
-    def load(self, collection_name, field_filters=None):
-        if field_filters is None:
-            field_filters = {}
+    def load(self, collection_name, query, selection):
 
         dbcollection = self.db[collection_name]
-        dbdocs = list(dbcollection.find(field_filters))
+        dbdocs = list(dbcollection.find(query, selection))
 
-        if len(dbdocs) == 0:
+        if len(dbdocs) is 0:
```

Comment:
Why switch from == to is?

Reply:
Technically `is` works on object comparison and `==` works on value comparison. I guess in this case it's a comparison between the lengths I was just trying to make the code to look more functional.

Reply:
I know the difference :) and I would agree if it was meant for `is False` but here it's really about integers.